src/Microsoft.Extensions.Logging.Abstractions/Internal/LogValuesFormatter.cs

```diff
@@ -18,24 +18,36 @@ public class LogValuesFormatter
         private const string NullValue = "(null)";
         private static readonly object[] EmptyArray = new object[0];
         private static readonly char[] FormatDelimiters = {',', ':'};
-        private readonly string _format;
+        private readonly Lazy<string> _format;
         private readonly List<string> _valueNames = new List<string>();
 
         public LogValuesFormatter(string format)
         {
             OriginalFormat = format;
+            _format = new Lazy<string>(() => FormatInput(format), true);
```

Comment:
2 additional allocations per format string.

Just make FormatInput check the _format field and calculate when it's null.

Reply:
Used this lazy loading logic as the variable was readonly. Refactored this code as per the comments and checked-in now.